source/common/http/conn_manager_impl.cc

```diff
@@ -1118,8 +1120,17 @@ void ConnectionManagerImpl::ActiveStream::setBufferLimit(uint32_t new_limit) {
   }
 }
 
-void ConnectionManagerImpl::ActiveStream::createFilterChain() {
-  connection_manager_.config_.filterFactory().createFilterChain(*this);
+bool ConnectionManagerImpl::ActiveStream::createFilterChain() {
+  auto upgrade = request_headers_->Upgrade();
+  if (upgrade != nullptr) {
+    if (!connection_manager_.config_.filterFactory().createUpgradeFilterChain(
+            upgrade->value().c_str(), *this)) {
```

Comment:
Is this already implemented? If not, a suggestion here: we need a way to yell filters that they are in upgrade filter chain and not http filter chain. 

Second, what happens if this returns false? Will connection fail? If so, won’t it break existing websocket behavior where users don’t have to define a filter chain?

Unlike http we don’t need a mandatory route filter here do we? Or we could have one and auto instantiate if not present.

Reply:
Yeah, we fail over to normal sendLocalReply behavior where we reject the request.  I'll add a comment here for clarity

Reply:
Oh and createUpgradeFilterChain was added in https://github.com/envoyproxy/envoy/pull/3685
I'm inclined to say if you configure a filter for upgrades and they need to know about upgrades they should check the header.